### PR TITLE
Validate reassign user in user delete command

### PR DESCRIPTION
**Pull request by ernilambar**
Fixes https://github.com/wp-cli/entity-command/issues/461

* This will perform sanity check for reassign value whether user is valid or not
* Asks for confirmation if not valid

### CODE REVIEW

**Comment by ernilambar** [comment]
> Can we add tests for this?

@swissspidy I could not find a way to add feature test for this because behat test pauses due to `WP_CLI::confirm()` here.

File: src/User_Command.php [hunk 1]
```diff
@@ -280,8 +280,10 @@ public function delete( $args, $assoc_args ) {
 			WP_CLI::error( 'Reassigning content to a different user is not supported on multisite.' );
 		}
 
-		if ( ! $reassign ) {
-			WP_CLI::confirm( '--reassign parameter not passed. All associated posts will be deleted. Proceed?', $assoc_args );
+		$is_reassign_valid = ( false === get_userdata( $reassign ) ) ? false : true;
+
+		if ( ! $reassign || ! $is_reassign_valid ) {
+			WP_CLI::confirm( '--reassign parameter not passed or invalid. All associated posts will be deleted. Proceed?', $assoc_args );
```
Comment (danielbachhuber) [inline src/User_Command.php:286]: I think we should actually have two separate confirmation prompts, so we can use more precise messages:

```php
if ( ! $is_reassign_valid ) {
	WP_CLI::confirm( '--reassign parameter is invalid. All associated posts will be deleted. Proceed?', $assoc_args );
}

if ( ! $reassign ) {
	WP_CLI::confirm( '--reassign parameter not passed. All associated posts will be deleted. Proceed?', $assoc_args );
}
```

Here's an example where session input is passed to the command:

https://github.com/wp-cli/entity-command/blob/fd187722baecc28b955b0a42d0b780042ca6e896/features/post-meta-clean-duplicates.feature#L33

You can follow a similar approach for writing a test here.

**Review by danielbachhuber — CHANGES_REQUESTED** [review]

File: src/User_Command.php (hunk 1 above)
Comment (ernilambar) [reply src/User_Command.php]: PR has been updated.

File: src/User_Command.php [hunk 2]
```diff
@@ -280,8 +280,12 @@ public function delete( $args, $assoc_args ) {
 			WP_CLI::error( 'Reassigning content to a different user is not supported on multisite.' );
 		}
 
+		$is_reassign_valid = ( false === get_userdata( $reassign ) ) ? false : true;
```
Comment (danielbachhuber) [inline src/User_Command.php:283]:
```suggestion
		$is_reassign_valid = ( $reassign && false === get_userdata( $reassign ) ) ? false : true;
```

Let's avoid the DB hit unless `$reassign` is provided.

**Review by danielbachhuber — APPROVED** [review]

**Comment by danielbachhuber** [comment]
Great work on this PR, @ernilambar !

**Pull request merged**